### ISSUE TITLE
ncu -i should execute npm install just once in a workspace.

### ISSUE THREAD

**Issue by ghdoergeloh** [raineorshine/npm-check-updates#1182]
- [x] I have searched for [similar issues](https://github.com/raineorshine/npm-check-updates/issues)
- [x] I am using the latest version of `npm-check-updates`
- [x] I am using `node >= 14`

---

## Steps to Reproduce

- node v16.15.1
- npm v8.11.0

Create the following structure:

```
├── package.json
└── packages
    ├── a
    │   └── package.json
    └── b
        └── package.json
```
root package.json:
```json
{
  "name": "ncu-bug",
  "workspaces": [
    "packages/**"
  ]
}
```

- Install e.g. `vanillajs@1.0.0` in "a" as well as in "b".
- Run `ncu -i -deep` in `ncu-bug`
- Accept every change
- Chhose "yes" if it asks to install.

## Current Behavior

```
Upgrading /path/to/package/ncu-bug/package.json

No dependencies.
Upgrading /path/to/package/ncu-bug/packages/a/package.json
[====================] 1/1 100%

✔ Choose which packages to update › 

 vanillajs  ^1.0.0  →  ^1.0.1
Upgrading /path/to/package/ncu-bug/packages/b/package.json
[====================] 1/1 100%

✔ Choose which packages to update › 

 vanillajs  ^1.0.0  →  ^1.0.1

✔ npm install in each project directory to install new versions? … yes

/home/georg/.nvm/versions/node/v16.15.1/lib/node_modules/npm-check-updates/build/src/index.js:56
    throw err;
    ^
npm ERR! code ENOTEMPTY
npm ERR! syscall rename
npm ERR! path /path/to/package/ncu-bug/node_modules/vanillajs
npm ERR! dest /path/to/package/ncu-bug/node_modules/.vanillajs-OvN89i9k
npm ERR! errno -39
npm ERR! ENOTEMPTY: directory not empty, rename '/path/to/package/ncu-bug/node_modules/vanillajs' -> '/path/to/package/ncu-bug/node_modules/.vanillajs-OvN89i9k'

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/georg/.npm/_logs/2022-08-24T12_55_19_273Z-debug-0.log

(Use `node --trace-uncaught ...` to show where the exception was thrown)
```

## Expected Behavior

It should run `npm install` only once in a workspace.

**Comment by sunnysingh** [comment]
I am also running into this, and as a quick fix it would be nice if a CLI option was available to avoid the auto-install prompt altogether. This way I can add a `package.json` script with something like this: `npm-check-updates --deep --interactive --no-install && npm install` (note the imaginary `--no-install` option). In the long term though, it would definitely be nice to have workspaces detected and the install command to work correctly so that I don't need to add the `&& npm install` command myself.

**Comment by raineorshine** [comment]
Workspace support added in prerelease `v16.2.0-0`. Open for feedback :). Your help is much appreciated in finding unidentified edge cases.

The API is slightly different than the OP in order to more closely match npm's `--workspaces` and `--workspace` options.

```
npx npm-check-updates@v16.2.0-0 --workspaces
```

## How it works

Upgrade all workspaces:

```
ncu --workspaces
ncu -ws
```

Upgrade a single workspace:

```
ncu --workspace a
ncu -w a
```

Upgrade more than one workspace:

```
ncu --workspace a --workspace b
ncu -w a -w b
```

Upgrade all workspaces AND the root project:

```
ncu --withWorkspaces
ncu -wws
```

Upgrade a single workspace AND the root project:

```
ncu --withWorkspace a
ncu -ww a
```

## Notes

- If run in `--interactive` mode, ncu will prompt to `npm install` once in the root project rather than separately in each workspace (#1182). 
- Running `--deep` will not trigger workspace support.

**Comment by sunnysingh** [comment]
Wow you work fast @raineorshine! This worked well for me, I confirmed that `--workspaces` does not upgrade the root project, and `--withWorkspaces` does. It seems to only run `npm install` within the root project, but I didn't do any extensive testing here. I will keep using the prerelease version and report back if I happen to come across any issues.

**Comment by raineorshine** [comment]
@httpete The only difference is the auto npm install in interactive mode.

- `npm exec --workspaces -- ncu -u -i` prompts to run `npm install` after each workspace is upgraded (once for each workspace).
- `npm --workspaces -i` prompts once at the end to run `npm install` in the project root.

**Comment by raineorshine** [comment]
Published to `latest` on `v16.3.0`. 

Note that `--withWorkspaces` has changed to a cleaner syntax of `--workspaces --root` (and `--withWorkspace` to `--workspace X --root`). Thanks to @mesqueeb for the suggestion.

**Comment by imguolao** [comment]
> Published to `latest` on `v16.3.0`.
> 
> Note that `--withWorkspaces` has changed to a cleaner syntax of `--workspaces --root` (and `--withWorkspace` to `--workspace X --root`). Thanks to @mesqueeb for the suggestion.

But it not working in v16.3.2. 

![image](https://user-images.githubusercontent.com/25502024/192702452-c1aa6385-2cb1-431a-bf18-95a20e151b5c.png)

**Comment by mesqueeb** [comment]
@raineorshine FYI `-ws` is shorthand for `--workspaces`, maybe you didn't enable this flag on the NCU side?

**Comment by raineorshine** [comment]
Flags are enabled and working as expected.

**Comment by apetta** [comment]
Is there a way to run on all except a select few workspaces when using `-ws`?

**Comment by raineorshine** [comment]
No, they would have to be specified explicitly with `-w`.
